Support minimal functionalities for Jazzy

Hyunsoo0o0 opened the pull request:
- Add `rmw_count_clients` and `rmw_count_services`
- Add `rmw_get_gid_for_client`
- Put unimplemented error in `rmw_take_dynamic_message`, `rmw_take_dynamic_message_with_info`, and `rmw_serialization_support_init`
- Include <cstdint> in types.cpp
- Ignore PRECONDITION_NOT_MET when checking WaitSet detach condition

Kumazuma commented on rmw_gurumdds_cpp/src/rmw_count.cpp:
```diff
@@ -97,4 +97,68 @@ rmw_count_subscribers(
+  rmw_ret_t ret = rmw_validate_full_topic_name(service_name, &validation_result, nullptr);
+  if (RMW_RET_OK != ret) {
+    return ret;
+  }
+  if (RMW_TOPIC_VALID != validation_result) {
```
```suggestion
  }
  
  if (RMW_TOPIC_VALID != validation_result) {
```

브라켓과 브라켓 사이는 띄워주세요. 다른 데도 개행해주세요

Commit:
Support minimal functionalities for Jazzy
- Add `rmw_count_clients` and `rmw_count_services`
- Add `rmw_get_gid_for_client`
- Put unimplemented error in `rmw_take_dynamic_message`, `rmw_take_dynamic_message_with_info`, and `rmw_serialization_support_init`
- Include <cstdint> in types.cpp
- Ignore PRECONDITION_NOT_MET when checking WaitSet detach condition

Commit:
Handle 'best available' QoS policies
